pkg/util/workloads_utils.go

```diff
@@ -294,6 +417,36 @@ func GetEmptyWorkloadObject(gvk schema.GroupVersionKind) client.Object {
 	}
 }
 
+func ParsePartitionFrom(object *unstructured.Unstructured) int32 {
```

Comment:
Why not make the parameter `object client.Object` so that you can use deployment, cloneset objects directly? Same question to other functions in this file.